File: examples/example-jdbc-spring/src/main/java/example/repositories/OwnerRepository.java
package example.repositories;

import java.util.List;
import java.util.Optional;

import org.springframework.data.repository.CrudRepository;

import example.domain.Owner;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;

@JdbcRepository(dialect = Dialect.H2)
public interface OwnerRepository extends CrudRepository<Owner, Long> {

    @Override
    List<Owner> findAll();

    Optional<Owner> findByName(String name);
}

File: examples/example-jdbc-kotlin/src/main/kotlin/example/repositories/OwnerRepository.kt
package example.repositories

import java.util.Optional

import example.domain.Owner
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository

@JdbcRepository(dialect = Dialect.H2)
interface OwnerRepository : CrudRepository<Owner, Long> {

    override fun findAll(): List<Owner>

    fun findByName(name: String): Optional<Owner>
}